### PR TITLE
Cosmos DB - Add PatchItem support

### PR DESCRIPTION
**Pull request by ealsur**
This is the follow-up of https://github.com/Azure/azure-sdk-for-go/pull/19440 - The original author cannot continue with the required public API changes so I am picking up their contribution and finalizing it.

API view: https://apiview.dev/Assemblies/Review/2e5e208d23f5462591095360cb741a86/d58b2a26917246daa56b0dbe986a9756?diffRevisionId=3c921922060940d9a8a7906608b7f05e&doc=False&diffOnly=True

Closes https://github.com/Azure/azure-sdk-for-go/issues/18682

### CODE REVIEW

**Comment by ealsur** [comment]
@jhendrixMSFT @JeffreyRichter the API view has been updated to address the comments from the previous PR:

https://apiview.dev/Assemblies/Review/2e5e208d23f5462591095360cb741a86/836fbb2cca6348bf9087730ec6dc1b5e?diffRevisionId=3c921922060940d9a8a7906608b7f05e&doc=False&diffOnly=True

File: sdk/data/azcosmos/cosmos_patch_operations.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	return buffer.Bytes(), nil
+}
+
+// SetCondition sets condition for the patch request.
+func (p *PatchOperations) SetCondition(condition string) {
+	p.condition = &condition
+}
+
+// AppendReplace appends a replace operation to the patch request.
+func (p *PatchOperations) AppendReplace(path string, value interface{}) {
```
Comment (jhendrixMSFT) [inline sdk/data/azcosmos/cosmos_patch_operations.go:69]: Nit: Go 1.18 introduced keyword `any` which is an alias for `interface{}` which we've been trying to use as it's shorter/cleaner.

Comment (ealsur) [reply sdk/data/azcosmos/cosmos_patch_operations.go]: Perfect. Done :)

File: sdk/data/azcosmos/cosmos_patch_operations.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ type PatchOperations struct {
 }
 
 // MarshalJSON implements the json.Marshaler interface
-func (o PatchOperations) MarshalJSON() ([]byte, error) {
+func (o *PatchOperations) MarshalJSON() ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (ealsur) [reply sdk/data/azcosmos/cosmos_patch_operations.go:43]: Currently the tests are passing, this Marshaller is working (the network request contains the required payload format). I can undo this but I thought this was a required change from the API View comment by Jeffrey

[the remark this answers is not in the document]
Comment (ealsur) [reply sdk/data/azcosmos/cosmos_patch_operations.go:43]: I changed this to `&ops` when passing it to be serialized by `azruntime.MarshalAsJSON` and it worked:

https://github.com/Azure/azure-sdk-for-go/blob/e37b03c257b89ae8e7b7da14b50a8ca20e0260ed/sdk/data/azcosmos/cosmos_container.go#L492-L498

[the remark this answers is not in the document]
Comment (ealsur) [reply sdk/data/azcosmos/cosmos_patch_operations.go:43]: Basically if the variable is a `&...` when `azruntime.MarshalAsJSON` is called, it uses the MarshalJSON with the `*`.

[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply sdk/data/azcosmos/cosmos_patch_operations.go:43]: The convention is that the receiver is passed by value for the MarshalJSON() method and is consistent with our other SDKs (and generated code).

https://pkg.go.dev/encoding/json

[the remark this answers is not in the document]
Comment (ealsur) [reply sdk/data/azcosmos/cosmos_patch_operations.go]: Reverted

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**